@stellar/typescript-wallet-sdk/src/walletSdk/Uri/sep7Parser.ts

```diff
@@ -0,0 +1,154 @@
+import { Sep7Pay, Sep7Tx } from "../Uri";
+import {
+  Sep7Replacement,
+  WEB_STELLAR_TX_SCHEME,
+  WEB_STELLAR_PAY_SCHEME,
+  Sep7OperationType,
+} from "../Types";
+import {
+  Sep7InvalidUriError,
+  Sep7UriTypeNotSupportedError,
+} from "../Exceptions";
+
+/**
+ * Returns true if the given URI is a SEP-7 compliant URI, false otherwise.
+ *
+ * Currently this checks whether it starts with 'web+stellar:tx' or 'web+stellar:pay'
+ * and has its required parameters: 'xdr=' and 'destination=' respectively.
+ *
+ * @param {string} uri The URI string to check.
+ *
+ * @returns {boolen} returns `true` if it's a valid Sep-7 uri, `false` otherwise.
+ */
+export const isValidSep7Uri = (uri: string): boolean => {
```

Comment:
Should this check that the `xdr` parameter for tx operations are a valid `TransactionEnvelope`?
Same question for `destination` on the pay operation, should it check that the value is a valid account ID?

Reply:
Good call, I like your suggestions :)

I think I'll make this function returning something like:
```
{
  result: boolean;
  reason: string;
}
```

Where `reason` should be a message letting devs know why the verification failed.

What do you think?

Reply:
yeah that sounds good to me, I like the more informational response. 👍

Reply:
Implemented here: https://github.com/stellar/typescript-wallet-sdk/pull/141/commits/64f6544abef3a0e5c7e5a271f4d189f18b093415
More tests to account for the error reason messages: https://github.com/stellar/typescript-wallet-sdk/pull/141/commits/89c504566d2d5b61ab7f874113b44c78f8182b95

Reply:
this lgtm 👏